pkg/k8s/labels.go

```diff
@@ -80,6 +84,7 @@ var proxyLabels = []string{
 	ProxyJobLabel,
 	ProxyDaemonSetLabel,
 	ProxyStatefulSetLabel,
+	ProxyPodLabel,
```

Comment:
Apologies for the conflict on this branch -- it stems from my changes in #1215. You just need to add the new label to the `podOwnerLabels` slice, instead of the `proxyLabels` slice.

Related to this change, the list of label consts here determines the set of labels that are returned by `GetOwnerLabels` func in this file. That func is called by the destination service, [here](https://github.com/runconduit/conduit/blob/master/controller/destination/listener.go#L130).

With your change, the list of labels returned will potentially include a "pod" key now, but we're already setting that label in the destination service:

```
				metricLabelsForPod = pkgK8s.GetOwnerLabels(pod.ObjectMeta)
				metricLabelsForPod["pod"] = pod.Name
```

I think it's highly unlikely that the pod name and the pod's pod name label will be different (unless somebody edits the label after inject), but I think we might want to protect against that. Maybe we should call the new label `k8s_pod` (like `k8s_job`) to prevent a conflict? Or maybe we should omit this label from the list of potential labels altogether, since it's guaranteed to be set by the destination service? I think I like the latter approach best, in which case I'd just revert adding the label to this list.